### ISSUE TITLE
Support creating PRs when Issues are disabled

### ISSUE THREAD

**Issue by sgianelli** [facebook/sapling#371]
I primarily work in a repository that has Github Issues disabled because we use other systems to track work and don't want to accidentally encourage the wrong behavior on our engineering team. This seems to result in me being unable create pull requests using `sl pr submit` with the following error:

```
abort: Error while trying to create a placeholder issue for a pull request on common-room/sparta: {
 "message": "Issues are disabled for this repo",
 "documentation_url": "https://docs.github.com/v3/issues/"
}
```

It would be nice if we didn't need to enable issues for our repo in order to use sapling.

**Comment by bolinfest** [comment]
Interesting. This is a result of an "improvement" we landed in the most recent release:

https://github.com/facebook/sapling/commit/e77e67bba710bc6665a4cc9119bc01303ff4509b

I suppose we can add a fallback path to get the previous behavior, but I think the new behavior is preferable when issues are supported.

**Comment by discentem** [comment]
I'd be interested in fallback behavior as well, with a config option to explicitly choose the old behavior. Whenever the new behavior is enabled, though, would it be possible for the placeholder issue to actually contain the commit message

**Comment by bolinfest** [comment]
@discentem Unfortunately, using the actual commit message with the placeholder doesn't really work out. For performance, the create requests are done in parallel. Only once we get the response are the numbers known and users [understandably] have shown a preference for stack order to match the numerical order. As such, we find out what placeholder numbers we have and *then* choose which commits to associate with each.

**Comment by bolinfest** [comment]
Taking a step back, if a repo has issues enabled, why would you prefer the old behavior?

**Comment by discentem** [comment]
Mostly the generic placeholder issues are annoying 😅 It seems like closing them has no undesired affect, perhaps they could be closed after the PRs are submitted?

**Comment by bolinfest** [comment]
I can see where they're annoying, but I'm curious how people are ending up with them sticking around. The code is written so that we convert the issue to a PR "right away," so are folks hitting ctrl-C inbetween the two API calls or unplugging their network cable?

Or is there a common logical error that causes a crash that we haven't seen on our side yet?

**Comment by discentem** [comment]
Oh, I misunderstood. The issues are not expected to stick around after getting converted to PRs and if they stick around it's considered a bug? I'll try to experiment more and figure out what conditions cause the issues to stay open.

**Comment by bolinfest** [comment]
@discentem indeed! If you are seeing that, it is definitely a bug. This behavior is hot of the presses and we just landed a test harness, so coverage is not great on our end yet, admittedly.
